torchgeo/datasets/utils.py

```diff
@@ -184,97 +200,134 @@ def download_radiant_mlhub_collection(
         Raises:
             ValueError: if bounding box is invalid
                 (minx > maxx, miny > maxy, or mint > maxt)
         """
-        if minx > maxx:
-            raise ValueError(f"Bounding box is invalid: 'minx={minx}' > 'maxx={maxx}'")
-        if miny > maxy:
-            raise ValueError(f"Bounding box is invalid: 'miny={miny}' > 'maxy={maxy}'")
-        if mint > maxt:
-            raise ValueError(f"Bounding box is invalid: 'mint={mint}' > 'maxt={maxt}'")
-
-        # Using super() doesn't work with mypy, see:
-        # https://stackoverflow.com/q/60611012/5828163
-        return tuple.__new__(cls, [minx, maxx, miny, maxy, mint, maxt])
-
-    def __init__(
-        self,
-        minx: float,
-        maxx: float,
-        miny: float,
-        maxy: float,
-        mint: float,
-        maxt: float,
-    ) -> None:
-        """Initialize a new instance of BoundingBox.
+        if self.minx > self.maxx:
+            raise ValueError(
+                f"Bounding box is invalid: 'minx={self.minx}' > 'maxx={self.maxx}'"
+            )
+        if self.miny > self.maxy:
+            raise ValueError(
+                f"Bounding box is invalid: 'miny={self.miny}' > 'maxy={self.maxy}'"
+            )
+        if self.mint > self.maxt:
+            raise ValueError(
+                f"Bounding box is invalid: 'mint={self.mint}' > 'maxt={self.maxt}'"
+            )
+
+    # https://github.com/PyCQA/pydocstyle/issues/525
+    @overload
+    def __getitem__(self, key: int) -> float:  # noqa: D105
+        pass
+
+    @overload
+    def __getitem__(self, key: slice) -> List[float]:  # noqa: D105
+        pass
+
+    def __getitem__(self, key: Union[int, slice]) -> Union[float, List[float]]:
+        """Index the (minx, maxx, miny, maxy, mint, maxt) tuple.
 
         Args:
-            minx: western boundary
-            maxx: eastern boundary
-            miny: southern boundary
-            maxy: northern boundary
-            mint: earliest boundary
-            maxt: latest boundary
+            key: integer or slice object
+
+        Returns:
+            the value(s) at that index
+
+        Raises:
+            IndexError: if key is out of bounds
         """
-        self.minx = minx
-        self.maxx = maxx
-        self.miny = miny
-        self.maxy = maxy
-        self.mint = mint
-        self.maxt = maxt
+        return [self.minx, self.maxx, self.miny, self.maxy, self.mint, self.maxt][key]
 
-    def __getnewargs__(self) -> Tuple[float, float, float, float, float, float]:
-        """Values passed to the ``__new__()`` method upon unpickling.
+    def __iter__(self) -> Iterator[float]:
+        """Container iterator.
 
         Returns:
-            tuple of bounds
+            iterator object that iterates over all objects in the container
         """
-        return self.minx, self.maxx, self.miny, self.maxy, self.mint, self.maxt
+        yield from [self.minx, self.maxx, self.miny, self.maxy, self.mint, self.maxt]
```

Comment:
Didn't even realize `yield from` was a thing until I had to figure out how to do this